pkg/services/baremetal/host/state_machine.go

```diff
@@ -142,6 +147,9 @@ func (hsm *hostStateMachine) updateSSHKey() actionResult {
 			}
 		}
 		if err := validateSSHKey(osSSHSecret, hsm.host.Spec.Status.SSHSpec.SecretRef); err != nil {
+			msg := fmt.Sprintf("ssh credentials for os are invalid: %s", err.Error())
```

Comment:
I would remove `for os`. But maybe add the hostname to the error message.

Reply:
Sorry, which hostname?

Reply:
I thought about this hostname of the `hostStateMachine`. But since the message will be in the condition of this hsm, it makes no sense. Ignore my comment :-)

Reply:
done